x-pack/plugins/enterprise_search/public/applications/app_search/components/engine/engine_router.tsx

```diff
@@ -93,12 +92,16 @@ export const EngineRouter: React.FC = () => {
           <AnalyticsRouter />
         </Route>
       )}
-      <Route path={ENGINE_DOCUMENT_DETAIL_PATH}>
-        <DocumentDetail />
-      </Route>
-      <Route path={ENGINE_DOCUMENTS_PATH}>
-        <Documents />
-      </Route>
+      {canViewEngineDocuments && (
+        <Route path={ENGINE_DOCUMENT_DETAIL_PATH}>
+          <DocumentDetail />
+        </Route>
+      )}
+      {canViewEngineDocuments && (
+        <Route path={ENGINE_DOCUMENTS_PATH}>
+          <Documents />
+        </Route>
+      )}
```

Comment:
- There isn't currently a role that would fail this check, but since the documents nav link is wrapped in the check, we might as well copy it and look at our role checks in the 8.0 refactor
- One alternative for a single conditional could be creating a single shared DocumentRouter view, but I didn't bother doing it. If people strongly prefer it I can, leave a comment if so